### PR TITLE
Use proforma gem

### PR DESCRIPTION
**Pull request by kkoehn**
Make use of new proforma-gem which handles the (de)serialization of Tasks

Contents of this PR:
- [x] add state tags to exercise (new / updated) set on import
  - [x] tag should be removable from index and show
- [x] add versioning of exercises (a new exercise with old values is created, on update and import - this exercise should not show in index)
  - [x] history is shown in show
- [x] add primary description (one of the descriptions of an exercise has to be primary)
  - [x] changable in edit
  - [x] validated
- [x] add export of exercise(s) to zip_file based on proforma
  - [x] support multiexport
- [x] add import of exercise(s) from zip_file based on proforma
  - [x] update source exercises, if applicable (create other otherwise) using uuids and checksums
- [x] add API export to arbitrary endpoint (Codeharbor)
  - [x] support CO's multistep import process
  - [ ] support multiexport (cart, collection) (later -> not in MVP)
  - [x] multistep import process (to check if uuid exists, with appropriate warnings)
- [x] add API-endpoint to import exercises from arbitrary sources (Codeocean)
  - [x] multistep import process (to check if uuid exists, with appropriate warnings)
    - [x] logic
    - [x] warnings
    - [x] new exercise should be "unpublished", which makes it unaccessible  (LTI token?)
  - [x] support CH's multistep import process

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 source 'https://rubygems.org'
 
+gem 'proforma', git: 'git://github.com/openHPI/proforma.git' # use version not master
```
[the remark this answers is not in the document]
Comment (kkoehn) [reply Gemfile]: We had not planned to publish the gem yet.
For now I'd just use the gem directly from the repo and use a tag to constrain the version
the downside is, that we have to update the gemfile everytime we update the gem

[the remark this answers is not in the document]
Comment (MrSerth) [reply Gemfile]: Alright, I am fine with that too.

**Comment by kkoehn** [comment]
@MrSerth No problem, thanks for all the reviews

**Comment by MrSerth** [comment]
I just updated my approval of this PR after reviewing the recent changes and would be happy to merge. Many thanks!

**Pull request merged**